utils/internal/download_gigaspeech_with_pyspeechcolab.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+# Copyright 2021  Jiayu Du
+#                 Seasalt AI, Inc (Author: Guoguo Chen)
+#                 Tsinghua University (Author: Shuzhou Chai)
+#                 Xiaomi Corporation (Author: Junbo Zhang)
+
+set -e
+set -o pipefail
+
+with_dict=False
+host=tsinghua
+subset={XL}
+
+. ./utils/parse_options.sh || exit 1
+
+if [ $# -ne 1 ]; then
+  echo "Usage: $0 <gigaspeech-dataset-dir>"
+  echo " e.g.: $0 /disk1/audio_data/gigaspeech"
+  echo ""
+  echo "This script downloads the entire GigaSpeech Dataset from Tsinghua host."
+  echo "We suggest having at least 1.0T of free space in the target directory."
+  echo "If dataset resources are updated, you can re-run this script for "
+  echo "incremental download."
+  exit 1
+fi
+
+gigaspeech_dataset_dir=$1
+mkdir -p $gigaspeech_dataset_dir || exit 1;
+
+# Check dependency
+python3 -c "import speechcolab" 2> /dev/null || \
+  (echo "$0: This recipe needs the package speechcolab installed.";
+   echo "To install:"
+   echo "    pip install speechcolab"; exit 1)
```

Comment:
I remember getting some issues with "||" and "( ... )". Did you make sure that this will be able to handle the error and properly print out the error message as well as the return code?

Reply:
I just tested it, and it looks works fine.